OrbitCore/BlockChain.h

```diff
@@ -2,229 +2,244 @@
-        block = block->m_Next;
-        ++count;
-      }
-
-      uint32_t index = a_Index % BlockSize;
-      return &block->m_Data[index];
+    uint32_t count = 1;
+    Block<T, BlockSize>* block = root_;
+    while (count * BlockSize < index && block && block->HasNext()) {
+      block = block->mutable_next();
+      ++count;
     }
 
-    return nullptr;
+    CHECK(block != nullptr);
+
+    return &block->Get(index % BlockSize);
   }
 
-  Block<T, BlockSize>* GetBlockContaining(const T* a_Element) {
-    Block<T, BlockSize>* block = m_Root;
-    while (block) {
-      uint32_t size = block->m_Size;
-      if (size) {
-        T* begin = &block->m_Data[0];
-        T* end = &block->m_Data[size - 1];
-        if (begin <= a_Element && end >= a_Element) {
-          return block;
-        }
-      }
-      block = block->m_Next;
+  T* GetElementAfter(const T* element) {
+    auto block = GetBlockContaining(element);
+    if (!block) {
+      return nullptr;
     }
 
-    return nullptr;
-  }
+    T* begin = &block->Get(0);
+    uint32_t index = element - begin;
 
-  T* GetElementAfter(const T* a_Element) {
-    auto block = GetBlockContaining(a_Element);
-    if (block) {
-      T* begin = &block->m_Data[0];
-      uint32_t index = a_Element - begin;
-      if (index < block->m_Size - 1)
-        return &block->m_Data[++index];
-      else if (block->m_Next && block->m_Next->m_Size)
-        return &block->m_Next->m_Data[0];
+    if (index + 1 < block->size()) {
+      return &block->Get(index + 1);
     }
-    return nullptr;
-  }
 
-  T* GetElementBefore(const T* a_Element) {
-    auto block = GetBlockContaining(a_Element);
-    if (block) {
-      T* begin = &block->m_Data[0];
-      uint32_t index = a_Element - begin;
-      if (index > 0)
-        return &block->m_Data[--index];
-      else if (block->m_Prev)
-        return &block->m_Prev->m_Data[block->m_Prev->m_Size - 1];
+    if (block->HasNext() && block->next()->size() != 0) {
+      return &block->mutable_next()->Get(0);
     }
+
     return nullptr;
   }
 
   BlockIterator<T, BlockSize> begin() {
-    return BlockIterator<T, BlockSize>(m_Root);
+    return BlockIterator<T, BlockSize>(root_);
   }
   BlockIterator<T, BlockSize> end() {
     return BlockIterator<T, BlockSize>(nullptr);
   }
 
-  Block<T, BlockSize>* m_Root;
-  Block<T, BlockSize>* m_Current;
-  uint32_t m_NumBlocks;
-  uint32_t m_NumItems;
+ private:
+  Block<T, BlockSize>* GetBlockContaining(const T* element) {
+    Block<T, BlockSize>* block = root_;
+    while (block != nullptr) {
+      uint32_t size = block->size();
+      if (size != 0) {
+        T* begin = &block->Get(0);
+        T* end = &block->Get(size - 1);
+        if (begin <= element && end >= element) {
+          return block;
+        }
+      }
+      block = block->mutable_next();
+    }
+
+    return nullptr;
+  }
+
+  Block<T, BlockSize>* root_;
+  Block<T, BlockSize>* current_;
+  uint32_t size_;
 };
+
+#endif  // ORBIT_CORE_BLOCK_CHAIN_H_
```

Comment:
Nit: Newline? Don't know if we have a "rule" for this, but Github shows it quite explicitly.

Reply:
I trust clang-format on this :)

Reply:
Fair enough